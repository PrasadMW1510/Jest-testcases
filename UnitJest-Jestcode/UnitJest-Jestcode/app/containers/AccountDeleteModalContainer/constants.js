/*
 *
 * AccountDeleteModalContainer constants
 *
 */

export const POST_ACCOUNT_DELETE_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST';
export const POST_ACCOUNT_DELETE_REQUEST_FAILURE =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST_FAILURE';
export const POST_ACCOUNT_DELETE_REQUEST_SUCCESS =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST_SUCCESS';
export const POST_ACCOUNT_DELETE_MIA_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_MIA_DELETE_REQUEST';
export const POST_ACCOUNT_UNENROLL_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST';
export const POST_ACCOUNT_UNENROLL_REQUEST_FAILURE =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST_FAILURE';
export const POST_ACCOUNT_UNENROLL_REQUEST_SUCCESS =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST_SUCCESS';
