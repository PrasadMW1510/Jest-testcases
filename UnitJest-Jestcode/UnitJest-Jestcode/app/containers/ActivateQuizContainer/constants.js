/*
 *
 * ActivateQuizContainer constants
 *
 */

export const ACTIVATE_QUIZ_LIST = 'app/ActivateQuizContainer/activate_quiz_list';
export const DEFAULT_ACTION = 'app/ActivateQuizContainer/DEFAULT_ACTION';
export const ACTIVATE_QUIZ_LIST_FAILURE = 'app/activateQuizContainer/activate-quiz-list-failure';
export const INITIALIZE_MODAL_FLAG = 'app/initialize-modal-flag';
