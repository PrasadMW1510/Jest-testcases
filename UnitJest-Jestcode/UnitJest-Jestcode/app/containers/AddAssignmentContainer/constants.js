/*
 *
 * AddAssignmentContainer constants
 *
 */

export const DEFAULT_ACTION = 'app/AddAssignmentContainer/DEFAULT_ACTION';
export const GET_STUDENTDETAIL_SUCCESS = 'app/AddAssignmentContainer/GET_STUDENTDETAIL_SUCCESS';
export const GET_STUDENTDETAIL_FAILURE = 'app/AddAssignmentContainer/GET_STUDENTDETAIL_FAILURE';
export const GET_STUDENT_DETAILS = 'app/AddAssignmentContainer/GET_STUDENT_DETAILS';
export const POST_NEW_ASSIGNMENT = 'app/AddAssignmentContainer/POST_NEW_ASSIGNMENT';
export const POST_NEW_ASSIGNMENT_SUCCESS = 'app/AddAssignmentContainer/POST_NEW_ASSIGNMENT_SUCCESS';
