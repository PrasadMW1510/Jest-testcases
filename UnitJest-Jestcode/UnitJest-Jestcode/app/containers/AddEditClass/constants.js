/*
 *
 * AddEditClass constants
 *
 */

export const SAVE_CLASS_REQUEST = 'AddEditClass/save_class_request';
export const SAVE_CLASS_REQUEST_SUCCESS = 'AddEditClass/save_class_request_success';
export const SAVE_CLASS_REQUEST_FAILURE = 'AddEditClass/save_class_request_failure';
export const SAVE_CLASS_MIA_REQUEST = 'AddEditClass/save_class_mia_request';
export const SAVE_CLASS_MIA_REQUEST_SUCCESS = 'AddEditClass/save_class_mia_request_success';
export const SAVE_CLASS_MIA_REQUEST_FAILURE = 'AddEditClass/save_class_mia_request_failure';

export const INITIALIZE_CLASS_FORM_REQUEST = 'AddEditClass/initialize_class_form_request';
export const INITIALIZE_CLASS_FORM_REQUEST_SUCCESS =
  'AddEditClass/initialize_class_form_request_success';
export const INITIALIZE_CLASS_FORM_REQUEST_FAILURE =
  'AddEditClass/initialize_class_form_request_failure';
export const META_DATA_GRADES = 'meta_data_grades';
export const META_DATA_TEACHERS = 'meta_data_teachers';
export const META_DATA_STUDENTS = 'meta_data_students';
export const FIELD_SELECTED_STUDENTS = 'rosterStudents';
export const FORM_CLASS_PROFILE = 'form_class_profile';
export const FORM_TITLE_ADD = 'Add a Class';
export const FORM_TITLE_EDIT = 'Edit Class Profile';
