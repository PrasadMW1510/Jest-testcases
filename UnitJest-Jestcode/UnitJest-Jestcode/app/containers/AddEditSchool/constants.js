/*
 *
 * AddEditSchool constants
 *
 */

export const FORM_SCHOOL_PROFILE = 'form_school_profile';
export const FORM_TITLE_ADD = 'Add a School';
export const FORM_TITLE_EDIT = 'Edit School Profile';

// Actions

export const META_DATA_GRADES = 'meta_data_grades';

export const GRADE_LIST_FOR_DISTRICT = 'AddEditSchool/grade_list_for_district';
export const GET_META_DATA_REQUEST = 'AddEditSchool/get_meta_data_request';
export const GET_META_DATA_REQUEST_SUCCESS = 'AddEditSchool/get_meta_data_request_success';
export const GET_META_DATA_REQUEST_FAILURE = 'AddEditSchool/get_meta_data_request_failure';

export const INITIALIZE_SCHOOL_FORM_REQUEST = 'AddEditSchool/initialize_school_form_request';
export const INITIALIZE_SCHOOL_FORM_REQUEST_SUCCESS =
  'AddEditSchool/initialize_school_form_request_success';
export const INITIALIZE_SCHOOL_FORM_REQUEST_FAILURE =
  'AddEditSchool/initialize_school_form_request_failure';

export const SAVE_SCHOOL_REQUEST = 'AddEditSchool/save_school_request';
export const SAVE_SCHOOL_REQUEST_SUCCESS = 'AddEditSchool/save_school_request_success';
export const SAVE_SCHOOL_REQUEST_FAILURE = 'AddEditSchool/save_school_request_failure';

export const SAVE_SCHOOL_MIA_REQUEST = 'AddEditSchool/save_school_mia_request';
