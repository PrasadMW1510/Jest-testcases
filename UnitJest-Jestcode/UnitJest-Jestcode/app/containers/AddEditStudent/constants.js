/*
 *
 * AddEditStudent constants
 *
 */

export const SAVE_STUDENT_REQUEST = 'AddEditStudent/save_student_request';
export const SAVE_STUDENT_REQUEST_SUCCESS = 'AddEditStudent/save_student_request_success';
export const SAVE_STUDENT_REQUEST_FAILURE = 'AddEditStudent/save_student_request_failure';
export const FORM_STUDENT_PROFILE = 'FORM_STUDENT_PROFILE';
export const INITIALIZE_STUDENT_FORM_REQUEST = 'AddEditStudent/initialize_student_form_request';
export const INITIALIZE_STUDENT_FORM_REQUEST_SUCCESS =
  'AddEditStudent/initialize_student_form_request_success';
export const INITIALIZE_STUDENT_FORM_REQUEST_FAILURE =
  'AddEditStudent/initialize_student_form_request_failure';
export const META_DATA_GRADES = 'meta_data_grades';
export const META_DATA_CLASSES = 'meta_data_classes';
export const META_DATA_GROUPS = 'meta_data_groups';
export const META_DATA_PASSWORD_CONFIG = 'meta_data_password_config';
export const FORM_TITLE_ADD = 'Add a Student';
export const FORM_TITLE_EDIT = 'Edit Student Profile';
