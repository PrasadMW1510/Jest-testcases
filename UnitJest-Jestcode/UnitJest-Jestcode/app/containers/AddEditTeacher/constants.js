/*
 *
 * AddEditTeacher constants
 *
 */

export const FORM_TEACHER_PROFILE = 'form_teacher_profile';

export const ADD_EDIT_TEACHER_REQUEST = 'app/AddEditTeacher/ADD_EDIT_TEACHER_REQUEST';
export const ADD_EDIT_TEACHER_REQUEST_SUCCESS =
  'app/AddEditTeacher/ADD_EDIT_TEACHER_REQUEST_SUCCESS';
export const ADD_EDIT_TEACHER_REQUEST_FAILURE =
  'app/AddEditTeacher/ADD_EDIT_TEACHER_REQUEST_FAILURE';

export const POST_ADD_TEACHER_REQUEST = 'app/AddEditTeacher/POST_ADD_TEACHER_REQUEST';
export const POST_ADD_TEACHER_REQUEST_SUCCESS =
  'app/AddEditTeacher/POST_ADD_TEACHER_REQUEST_SUCCESS';
export const POST_ADD_TEACHER_REQUEST_FAILURE =
  'app/AddEditTeacher/POST_ADD_TEACHER_REQUEST_FAILURE';

export const POST_SAVE_TEACHER_REQUEST = 'app/AddEditTeacher/POST_SAVE_TEACHER_REQUEST';
export const POST_SAVE_TEACHER_REQUEST_SUCCESS =
  'app/AddEditTeacher/POST_SAVE_TEACHER_REQUEST_SUCCESS';
export const POST_SAVE_TEACHER_REQUEST_FAILURE =
  'app/AddEditTeacher/POST_SAVE_TEACHER_REQUEST_FAILURE';

export const POST_SAVE_TEACHER_MIA_REQUEST = 'app/AddEditTeacher/POST_SAVE_TEACHER_MIA_REQUEST';

export const ADD_EDIT_TEACHER_REQUEST_PROFILE_DETAILS_SUCCESS =
  'app/AddEditTeacher/ADD_EDIT_TEACHER_REQUEST_PROFILE_DETAILS_SUCCESS';
export const ADD_EDIT_TEACHER_REQUEST_PERMISSIONS_SUCCESS =
  'app/AddEditTeacher/ADD_EDIT_TEACHER_REQUEST_PERMISSIONS_SUCCESS';
export const ADD_EDIT_TEACHER_REQUEST_SCHOOL_AND_CLASSES_SUCCESS =
  'app/AddEditTeacher/ADD_EDIT_TEACHER_REQUEST_SCHOOL_AND_CLASSES_SUCCESS';
export const ADD_EDIT_TEACHER_REQUEST_PASSWORD_CONFIG_SUCCESS =
  'app/AddEditTeacher/ADD_EDIT_TEACHER_REQUEST_PASSWORD_CONFIG_SUCCESS';
