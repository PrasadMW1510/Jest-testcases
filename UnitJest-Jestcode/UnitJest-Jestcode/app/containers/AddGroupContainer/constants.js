/*
 *
 * AddGroupContainer constants
 *
 */

export const GET_CLASSES_WITH_STUDENT_INFO = 'get_classes_with_student_info';

export const GET_CLASSES_WITH_STUDENT_INFO_SUCCESS = 'get_classes_with_student_info_success';

export const GET_CLASSES_WITH_STUDENT_INFO_FAILURE = 'get_classes_with_student_info_failure';

export const POST_GROUP = 'post_group';

export const POST_GROUP_SUCCESS = 'post_group_success';

export const POST_GROUP_FAILURE = 'post_group_failure';

export const RESET_GROUP_STATUS = 'reset_group_status';

export const GET_GROUP_INFO = 'get_group_info';

export const GET_GROUP_INFO_SUCCESS = 'get_group_info_success';

export const GET_GROUP_INFO_FAILURE = 'get_group_info_failure';

export const GROUP_TITLE = {
  add: 'Add a Group',
  edit: 'Edit Group Profile',
};
