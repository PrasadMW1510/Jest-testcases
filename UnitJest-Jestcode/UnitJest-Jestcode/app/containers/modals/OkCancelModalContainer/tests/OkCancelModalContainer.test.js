import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { OK_CANCEL_MODAL } from 'containers/ModalController/constants';
import { OkCancelModalContainer } from '../index';

describe('<OkCancelModalContainer />', () => {
  let wrapper = null;
  let mockData = null;
  const mockHideModal = jest.fn();
  const mockOnCancelFunc = jest.fn();
  const mockOnCancelParam = { key: 'cancel' };
  const mockOnOkFunc = jest.fn();
  const mockOnOkParam = { key1: 'ok' };
  const onClickAction = { preventDefault: jest.fn() };

  describe('Modal with all overrides', () => {
    beforeEach(() => {
      mockData = {
        heading: 'This is a heading',
        onCancel: mockOnCancelFunc,
        onCancelParam: mockOnCancelParam,
        onOk: mockOnOkFunc,
        onOkParam: mockOnOkParam,
      };
      wrapper = shallow(
        <OkCancelModalContainer
          cancelLabel="No"
          data={mockData}
          headerClassName="header-class-name"
          hideModal={mockHideModal}
          modalClassName="modal-class-name"
          okLabel="Yes"
        />
      );
    });

    it('Expect to render label overrides correctly', () => {
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('Expect handleOk to execute correctly', () => {
      wrapper
        .find('SAMButton')
        .first()
        .prop('onClickHandler')(onClickAction);
      expect(mockOnOkFunc).toHaveBeenCalledWith(mockOnOkParam);
      expect(mockHideModal).toHaveBeenCalledWith(OK_CANCEL_MODAL, {
        buttonClicked: 'ok',
        ...mockData,
      });
    });

    it('Expect handleCancel to execute correctly', () => {
      wrapper
        .find('SAMButton')
        .last()
        .prop('onClickHandler')(onClickAction);
      expect(mockOnCancelFunc).toHaveBeenCalledWith(mockOnCancelParam);
      expect(mockHideModal).toHaveBeenCalledWith(OK_CANCEL_MODAL, {
        buttonClicked: 'cancel',
        ...mockData,
      });
    });
  });

  describe('Modal with all defaults', () => {
    beforeEach(() => {
      mockData = {};
      wrapper = shallow(<OkCancelModalContainer data={mockData} hideModal={mockHideModal} />);
    });

    it('Expect to render label overrides correctly', () => {
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('Expect handleOk to execute correctly', () => {
      wrapper
        .find('SAMButton')
        .first()
        .prop('onClickHandler')(onClickAction);
      expect(mockHideModal).toHaveBeenCalledWith(OK_CANCEL_MODAL, {
        buttonClicked: 'ok',
        ...mockData,
      });
    });

    it('Expect handleCancel to execute correctly', () => {
      wrapper
        .find('SAMButton')
        .last()
        .prop('onClickHandler')(onClickAction);
      expect(mockHideModal).toHaveBeenCalledWith(OK_CANCEL_MODAL, {
        buttonClicked: 'cancel',
        ...mockData,
      });
    });
  });
});
