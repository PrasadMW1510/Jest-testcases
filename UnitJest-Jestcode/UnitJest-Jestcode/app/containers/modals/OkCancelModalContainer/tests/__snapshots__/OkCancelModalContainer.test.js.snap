// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OkCancelModalContainer /> Modal with all defaults Expect to render label overrides correctly 1`] = `
<SAMModal
  contentLabel="SAM Modal"
  isOpen={true}
  modalClassModifier="ok-cancel-modal "
>
  <div
    className="ok-cancel-modal--body"
  >
    <div
      className="ok-cancel-modal--message"
    />
    <SAMButton
      buttonClassModifier="ok-cancel-modal--ok-button"
      buttonType="button"
      isPrimaryButton={true}
      onClickHandler={[Function]}
    >
      OK
    </SAMButton>
    <SAMButton
      buttonClassModifier="ok-cancel-modal--cancel-button"
      buttonType="button"
      isPrimaryButton={false}
      onClickHandler={[Function]}
    >
      Cancel
    </SAMButton>
  </div>
</SAMModal>
`;

exports[`<OkCancelModalContainer /> Modal with all overrides Expect to render label overrides correctly 1`] = `
<SAMModal
  contentLabel="SAM Modal"
  isOpen={true}
  modalClassModifier="ok-cancel-modal "
>
  <h2
    className="ok-cancel-modal--header "
  >
    This is a heading
  </h2>
  <div
    className="ok-cancel-modal--body"
  >
    <div
      className="ok-cancel-modal--message"
    />
    <SAMButton
      buttonClassModifier="ok-cancel-modal--ok-button"
      buttonType="button"
      isPrimaryButton={true}
      onClickHandler={[Function]}
    >
      OK
    </SAMButton>
    <SAMButton
      buttonClassModifier="ok-cancel-modal--cancel-button"
      buttonType="button"
      isPrimaryButton={false}
      onClickHandler={[Function]}
    >
      Cancel
    </SAMButton>
  </div>
</SAMModal>
`;
